homeassistant/components/weather/__init__.py

```diff
@@ -110,6 +118,10 @@ def state_attributes(self):
         if attribution is not None:
             data[ATTR_WEATHER_ATTRIBUTION] = attribution
 
+        forecast = self.forecast
+        if forecast is not None:
+            data[ATTR_FORECAST] = forecast
```

Comment:
I think this should be done in the base weather component and not in the platforms. I suggest converting the property `_temp_for_display` to an internal method that accepts a temperature as argument. This method can then be used both for converting the current temperature and the forecast temperature in the state attributes. One problem with this might be that we need to know the data structure of the forecast to be able to access the temp data of the forecast for conversion, from all platforms. But maybe this is already required for other modules like the frontend?